src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DefaultDockerClient.java

```diff
@@ -166,6 +167,16 @@ private Optional<UserNotAuthorizedForDockerException> checkUserAuthorizationErro
         return Optional.ofNullable(error);
     }
 
+    public void deleteImage(Image image) throws DockerImageDeleteException {
+        CliResponse response = runDockerCmd(String.format("docker rmi -f %s", image.getImageFullName()));
+        if (response.exit.isPresent() && response.exit.get() == 0) {
+            return;
+        } else {
+            throw new DockerImageDeleteException("Unexpected error while trying to perform docker rmi",
```

Comment:
if it is failing, we need to log the reason why it failed so that we and the customer can debug the issue. Log the standard out and standard error response from the command execution.